panther_analysis_tool/main.py

```diff
@@ -442,31 +503,55 @@ def confirm_analysis_exists(args: argparse.Namespace, analysis_id_list: list) ->
     return analysis_id_list
 
 
-# pylint: disable=too-many-locals
-def delete_analysis(args: argparse.Namespace) -> Tuple[int, str]:
-
+def delete_queries(args: argparse.Namespace, query_list: list) -> Tuple[int, str]:
     client = get_client(args.aws_profile, "lambda")
-    analysis_id_list = args.analysis_id
-    payload: dict = {"deleteDetections": {"entries": []}}
 
-    # Validate the Detection that we are deleting exists in Panther
-    # If nothing was found bail out, otherwise prepare to delete what has been confirmed
+    datalake_function = "panther-snowflake-api"
+    if args.athena_datalake:
+        datalake_function = "panther-athena-api"
 
-    analysis_id_list = confirm_analysis_exists(args, analysis_id_list)
-    if len(analysis_id_list) == 0:
-        logging.error("No matching analysis found, exiting")
-        return 1, ""
+    # Delete function needs the query ID, required endpoint wont take a list
+    query_id_list = []
+    for query in query_list:
+        payload = {"listSavedQueries": {"pageSize": 1, "name": query}}
+        list_response = client.invoke(
+            FunctionName=datalake_function,
+            InvocationType="RequestResponse",
+            LogType="None",
+            Payload=json.dumps(payload),
+        )
+        api_response = json.loads(list_response["Payload"].read().decode("utf-8"))
+        query_id = api_response.get("savedQueries")[0]["id"]
```

Comment:
will this be 'ok' if no queries are returned?

Reply:
Good call, I think it would probably explode. The API response is an empty `savedQueries` list if it can't find the query you searched for. I added some error checking that should keep it from detonating